### PR TITLE
Improvements to `deploy` and `preview` commands

### LINKED ISSUE #1: Command `infra:preview` fails if using local pulumi storage
If you are using local pulumi storage (e.g. `pulumi login --local`), running the `infra:preview` command results in the following error: 

```shell
$ twilio infra:preview --stack dev
error: getting secrets manager: passphrase must be set with PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE environment variables
```

If you run it through the Pulumi CLI directly, this will prompt you for the passphrase: 

```shell
$ pulumi preview --stack dev
Enter your passphrase to unlock config/secrets
    (set PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE to remember): 
```

### LINKED ISSUE #2: Command `infra:deploy` fails if using local pulumi storage
this is similar to #1 but for the `deploy` command

### PR DESCRIPTION
**Pull request by vernig**
When using Pulumi with local service, a passphrase needs to be generated and used for each command.
Since the pulumi CLI is executed in non interactive mode, we need to pass the passphrase as env variable.

As first step, the code is checking if Pulumi is running with local service (using `pulumi login` output). If it does, it prompts the user to provide the passphrase and add that to the env variables

Fix #1
Fix #2

### CODE REVIEW

**Comment by vernig** [comment]
@kaiquelupo can you merge the PRs from https://github.com/kaiquelupo/plugin-twilio-infra so you can test this changes as well?

File: src/utils.js [hunk 1]
```diff
@@ -130,20 +138,58 @@ function setStack(flags, args, clientCommand) {
...
-
-  shell.exec(`${vars} ${command} --stack=${stackName} ${commandFlags || ""}`).stdout;
+  let vars = getEnvironmentVariables(flags, twilioClient, stackName);
+
+  const pulumiCLI = await checkPulumi();
+  if (pulumiCLI.local) {
+    const answers = await inquirer.prompt([
+      {
+        type: "password",
+        name: "passPhrase",
```
Comment (bermudezmt) [inline src/utils.js]: Could this in theory also work if we changed the name to `passphrase` and `answers.passphrase`? I was just wondering why it's using camelcase.

**Review by bermudezmt — APPROVED** [review]
I can confirm that it works, but we should probably actually set the passphrase as an environment variable after entering it on stack initialization. Right now, every invocation of a twilio infra subcommand requires a passphrase when using local storage.

Thanks for your work, @vernig !

File: src/utils.js (hunk 1 above)
Comment (vernig) [reply src/utils.js]: Good point @bermudezmt . Done in e1a6f3e

**Comment by vernig** [comment]
> I can confirm that it works, but we should probably actually set the passphrase as an environment variable after entering it on stack initialization. Right now, every invocation of a twilio infra subcommand requires a passphrase when using local storage.

This is a very good point @bermudezmt. I fixed it in https://github.com/twilio-infra-as-code/plugin-twilio-infra/pull/3/commits/dbeafaa04424c2e251c1ad933c0d0c4b148209e9 but didn't thoroughly tested yet. Let me know if you get a chance to do it. 

Also, I added some more proper error handling and made the `--stack` flag required for `preview` and `deploy`

**Pull request merged**